### PR TITLE
Added Linked Video functionality

### PR DESCRIPTION
**Pull request by pk1m**
Currently, you are able to submit youtube or dailymotion valid video links and it will create a page for it with the embedded videos, caption and title.  Comments are also able to be added.

### CODE REVIEW

File: app/models/LinkedVideo.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  
+  protected String getEmbedHtml(){
+    if(link_type == 'y')
+      return "<iframe class=\"youtube-player\" type=\"text/html\" width=\"560\" height=\"340\" src=\"http://www.youtube.com/embed/" + video_id +  "\" frameborder=\"0\"> </iframe>";
+    else if (link_type == 'd')
+      return "<iframe src=\"http://www.dailymotion.com/embed/video/" + video_id + "/\" width=\"560\" height=\"340\" frameborder=\"0\"></iframe>";
+    else if(link_type == 'v')
+      return "";
+    else
+      return "NOOOO";
```
Comment (amshali) [inline app/models/LinkedVideo.java]: NOOOO? What does it mean?

File: app/models/LinkedVideo.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package models;
+
+import java.util.*;
+import javax.persistence.*;
+import play.db.jpa.*;
+import play.data.validation.*;
+import java.util.regex.*;
+
+@Entity
+public class LinkedVideo extends Commentable {
```
Comment (amshali) [inline app/models/LinkedVideo.java]: This must inherit from Post and shows up in the main page.

**Comment by amshali** [comment]
@jtessler, how do you post photo to the news page? Video must be posted to the new page and I believe they must inherit from Post. What you think?

**Comment by jtessler** [comment]
Yes, that sounds right. The news feed simply returns all `Post` models you're allowed to see, meaning all we need to do it extend from `Post`.

**Comment by amshali** [comment]
Sounds like that to me too.

**Comment by amshali** [comment]
Please try pulling and pushing again. I would say save your changes somewhere. Get a clone and then apply your changes back and push from there.

**Comment by amshali** [comment]
Any progress here?

**Comment by ProCynic** [comment]
You're going to need to make some changes to deal with what I did in #129.  If this inherits from Post, then it also inherits from Likeable, which means that owner has been pulled up into a super class.  So yo're going to need a constructor

```
public Video(User owner, ...) {
  super(owner);  //Can also pass in visibility here, will default to friends otherwise.
  ...
}
```

**Comment by pk1m** [comment]
I thought it was good after I merged it.  Now the main branch is updated so should I just open a new merge request, pulling from the main branch is giving a lot of merging problems.

**Comment by pk1m** [comment]
Can I merge this yet? 

And also ProCynic is this line not sufficient?
super(owner, owner, "");

My Videos file isn't currently finished yet so its not being used. Rather LinkedVideo is the one thats completed.

**Comment by ProCynic** [comment]
That's right if you extend from post.
-Geoffrey Parker

On Wed, Apr 25, 2012 at 12:34 AM, pk1m <
reply@reply.github.com

**Comment by amshali** [comment]
Let me test it and then I'll merge it in 2 hours.

File: app/models/Timeline.java.BACKUP.16455.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package models;
```
Comment (amshali) [inline app/models/Timeline.java.BACKUP.16455.java]: This file is strange. I am pretty sure it must not be here. Please remove it.

File: app/models/Timeline.java.BASE.16455.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package models;
```
Comment (amshali) [inline app/models/Timeline.java.BASE.16455.java]: This file is strange. I am pretty sure it must not be here. Please remove it.

File: app/models/Timeline.java.REMOTE.16455.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package models;
```
Comment (amshali) [inline app/models/Timeline.java.REMOTE.16455.java]: This file is strange. I am pretty sure it must not be here. Please remove it.

**Comment by amshali** [comment]
The bootstrap.zip file shouldn't be in this commit?!

**Comment by amshali** [comment]
Please fix the issues I mentioned here and we can go ahead and merge it then.

**Comment by pk1m** [comment]
I'll remove those but I don't think I added those myself, or at least purposefully.

**Comment by amshali** [comment]
@ProCynic , Do you know why this bootstrap.css is being added here? (along with a couple of more bootstrap files)

**Comment by ProCynic** [comment]
Looks like you added a couple icon files?  bootstrap.zip is an archive of all the assets that gets automatically generated when you make bootstrap.  We're not using it so it doesn't matter.

**Comment by amshali** [comment]
@pk1m , bootstrap.css is an empty file. Please remove it.

**Comment by ProCynic** [comment]
what? We need /public/bootstrap/docs/assets/css/bootstrap.css.  That's the bootstrap css file.  I think you tried to remake bootstrap and it didn't go well.  **Do not merge until you fix this!**

**Comment by pk1m** [comment]
I haven't touched anything relating to bootstrap or any assets for that matter.  Should I just delete it?

**Comment by ProCynic** [comment]
no.  somehow you deleted the contents of the bootstrap.css file, which we
need.  undo that.
-Geoffrey Parker

On Wed, Apr 25, 2012 at 7:18 PM, pk1m <
reply@reply.github.com

**Comment by ProCynic** [comment]
Did you test this at all? It doesn't even compile!

**Comment by amshali** [comment]
It turned out that your code had some compile error. Please pull, make your changes and make sure it compiles and works and push again.

**Comment by ProCynic** [comment]
The video class extends commentable but doesn't take into account the
changes I made to likeable a couple days ago.
-Geoffrey Parker

On Wed, Apr 25, 2012 at 8:25 PM, Amin Shali <
reply@reply.github.com

**Comment by ProCynic** [comment]
to fix the missing method for markdown, run play dependencies.
-Geoffrey Parker

On Wed, Apr 25, 2012 at 11:06 PM, pk1m <
reply@reply.github.com

**Comment by pk1m** [comment]
Seems like the changes have broken my code.  I'm getting a persistenceException saying It could not insert  LinkedVideo, so my .save() is not being allowed to go through when I submit a video.

Caused by: org.h2.jdbc.JdbcSQLException: Invalid value "13" for parameter "parameterIndex"

Could this be related to any of the new changes?  

I'll keep trying to figure it out.

**Comment by amshali** [comment]
Could you please provide a full stack trace and tell me what are you trying to do when this happens?

**Comment by pk1m** [comment]
23:19:24,915 ERROR ~ Invalid value "13" for parameter "parameterIndex" [90008-149]
23:19:24,976 ERROR ~ 

@6a6nedpj4
Internal Server Error (500) for request POST /videos

Execution exception (In /app/controllers/Videos.java around line 63)
PersistenceException occured : org.hibernate.exception.GenericJDBCException: could not insert: [models.LinkedVideo]

play.exceptions.JavaExecutionException: org.hibernate.exception.GenericJDBCException: could not insert: [models.LinkedVideo]
    at play.mvc.ActionInvoker.invoke(ActionInvoker.java:231)
    at Invocation.HTTP Request(Play!)
Caused by: javax.persistence.PersistenceException: org.hibernate.exception.GenericJDBCException: could not insert: [models.LinkedVideo]
    at org.hibernate.ejb.AbstractEntityManagerImpl.convert(AbstractEntityManagerImpl.java:1214)
    at org.hibernate.ejb.AbstractEntityManagerImpl.convert(AbstractEntityManagerImpl.java:1147)
    at org.hibernate.ejb.AbstractEntityManagerImpl.convert(AbstractEntityManagerImpl.java:1153)
    at org.hibernate.ejb.AbstractEntityManagerImpl.persist(AbstractEntityManagerImpl.java:678)
    at play.db.jpa.JPABase._save(JPABase.java:25)
    at play.db.jpa.GenericModel.save(GenericModel.java:203)
    at controllers.Videos.createLinkedVideo(Videos.java:63)
    at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
    at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
    at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
    at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
    at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
    ... 1 more
Caused by: org.hibernate.exception.GenericJDBCException: could not insert: [models.LinkedVideo]
    at org.hibernate.exception.SQLStateConverter.handledNonSpecificException(SQLStateConverter.java:140)
    at org.hibernate.exception.SQLStateConverter.convert(SQLStateConverter.java:128)
    at org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:66)
    at org.hibernate.id.insert.AbstractReturningDelegate.performInsert(AbstractReturningDelegate.java:64)
    at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2345)
    at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2852)
    at org.hibernate.action.EntityIdentityInsertAction.execute(EntityIdentityInsertAction.java:71)
    at org.hibernate.engine.ActionQueue.execute(ActionQueue.java:273)
    at org.hibernate.event.def.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:320)
    at org.hibernate.event.def.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:203)
    at org.hibernate.event.def.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:129)
    at org.hibernate.ejb.event.EJB3PersistEventListener.saveWithGeneratedId(EJB3PersistEventListener.java:69)
    at org.hibernate.event.def.DefaultPersistEventListener.entityIsTransient(DefaultPersistEventListener.java:179)
    at org.hibernate.event.def.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:135)
    at org.hibernate.event.def.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:61)
    at org.hibernate.impl.SessionImpl.firePersist(SessionImpl.java:808)
    at org.hibernate.impl.SessionImpl.persist(SessionImpl.java:782)
    at org.hibernate.impl.SessionImpl.persist(SessionImpl.java:786)
    at org.hibernate.ejb.AbstractEntityManagerImpl.persist(AbstractEntityManagerImpl.java:672)
    ... 9 more
Caused by: org.h2.jdbc.JdbcSQLException: Invalid value "13" for parameter "parameterIndex" [90008-149]
    at org.h2.message.DbException.getJdbcSQLException(DbException.java:327)
    at org.h2.message.DbException.get(DbException.java:167)
    at org.h2.message.DbException.getInvalidValueException(DbException.java:213)
    at org.h2.jdbc.JdbcPreparedStatement.setParameter(JdbcPreparedStatement.java:1259)
    at org.h2.jdbc.JdbcPreparedStatement.setString(JdbcPreparedStatement.java:323)
    at org.hibernate.type.descriptor.sql.VarcharTypeDescriptor$1.doBind(VarcharTypeDescriptor.java:52)
    at org.hibernate.type.descriptor.sql.BasicBinder.bind(BasicBinder.java:91)
    at org.hibernate.type.AbstractStandardBasicType.nullSafeSet(AbstractStandardBasicType.java:282)
    at org.hibernate.type.AbstractStandardBasicType.nullSafeSet(AbstractStandardBasicType.java:277)
    at org.hibernate.type.AbstractSingleColumnStandardBasicType.nullSafeSet(AbstractSingleColumnStandardBasicType.java:85)
    at org.hibernate.persister.entity.AbstractEntityPersister.dehydrate(AbstractEntityPersister.java:2184)
    at org.hibernate.persister.entity.AbstractEntityPersister.dehydrate(AbstractEntityPersister.java:2161)
    at org.hibernate.persister.entity.AbstractEntityPersister$4.bindValues(AbstractEntityPersister.java:2339)
    at org.hibernate.id.insert.AbstractReturningDelegate.performInsert(AbstractReturningDelegate.java:56)
    ... 24 more
^C~ ...

Occurs when I go to Videos under favorites and try to add a new LinkedVideo. When I click submit this exception pops up.  Last time I had a similar error when I came to you, one of my attributes was a char and I think the DB didn't like that.

**Pull request merged**